src/query.c

```diff
@@ -916,6 +1011,64 @@ static IndexIterator *Query_EvalTagPrefixNode(QueryEvalCtx *q, TagIndex *idx, Qu
   return NewUnionIterator(its, itsSz, q->docTable, 1, weight, QN_PREFIX, qn->pfx.tok.str);
 }
 
+/* Evaluate a tag prefix by expanding it with a lookup on the tag index */
+static IndexIterator *Query_EvalTagWildcardNode(QueryEvalCtx *q, TagIndex *idx, QueryNode *qn,
+                                              IndexIteratorArray *iterout, double weight) {
+  if (qn->type != QN_WILDCARD_QUERY) {
+    return NULL;
+  }
+  if (!idx || !idx->values) return NULL;
+
+  qn->verb.tok.len = Wildcard_RemoveEscape(qn->verb.tok.str, qn->verb.tok.len);
+  RSToken *tok = &qn->verb.tok;
+
+  size_t itsSz = 0, itsCap = 8;
+  IndexIterator **its = rm_calloc(itsCap, sizeof(*its));
+
+  // brute force wildcard query 
+  TrieMapIterator *it = TrieMap_Iterate(idx->values, tok->str, tok->len);
+  TrieMapIterator_SetTimeout(it, q->sctx->timeout);
+  // If there is no '*`, the length is known which can be used for optimization
+  it->mode = strchr(tok->str, '*') ? TM_WILDCARD_MODE : TM_WILDCARD_FIXED_LEN_MODE;
+
+  // an upper limit on the number of expansions is enforced to avoid stuff like "*"
+  char *s;
+  tm_len_t sl;
+  void *ptr;
+
+  // Find all completions of the prefix
+  while (TrieMapIterator_NextWildcard(it, &s, &sl, &ptr) &&
+        (itsSz < RSGlobalConfig.maxPrefixExpansions)) {
+          // TODO: use NewIndexReaderGeneric
+    //NewIndexReaderGeneric(ptr, q->sctx->spec, )
+    IndexIterator *ret = TagIndex_OpenReader(idx, q->sctx->spec, s, sl, 1);
+    if (!ret) continue;
+
+    // Add the reader to the iterator array
+    its[itsSz++] = ret;
```

Comment:
Can we use the arr.h here?

Reply:
We use the same logic on all iterators above. IMO, no need to change here.